source/features/link-to-changelog-file.tsx

```diff
@@ -10,9 +10,10 @@ import * as api from '../github-helpers/api';
 import {buildRepoURL, getRepo} from '../github-helpers';
 
 const getCacheKey = (): string => `changelog:${getRepo()!.nameWithOwner}`;
+const changelogNames = ['changelog', 'news', 'changes', 'history', 'release', 'whatsnew'];
 
 function parseFromDom(): false {
-	void cache.set(getCacheKey(), select('.js-navigation-item [title^="changelog" i]')?.textContent ?? false);
+	void cache.set(getCacheKey(), select(changelogNames.map(filename => `.js-navigation-item .js-navigation.open[title^="${filename}" i]`).join())?.textContent ?? false);
```

Comment:
Made the selector more specific because the commit messages in the file list can get selected by `.js-navigation-item [title^="changelog" i]`, e.g. on https://github.com/sphinx-doc/sphinx:
```html
<a data-pjax="true" title="changelog: split off older changes (pre-1.0) to a CHANGES.old file" class="link-gray" href="/sphinx-doc/sphinx/commit/b0381f97f877f9c3c0d005dbabde24a4fd979714">changelog: split off older changes (pre-1.0) to a CHANGES.old file</a>
```

Reply:
I’d expect `navigation` to always be inside `navigation-item` anyway. Does this work? Also select accepts an array as well

```suggestion
	void cache.set(getCacheKey(), select(changelogNames.map(filename => `.js-navigation.open[title^="${filename}" i]`))?.textContent ?? false);
```

Reply:
The API function below ensures that only the basename is checked, but this one doesn’t.
maybe it’s easier to change this whole thing to:

```
const filename = select.all(files).map(=>filename).find(same condition as the API function)
```

or you can use a for-of